pint/models/model_builder.py

```diff
@@ -70,8 +70,15 @@ def preprocess_parfile(self, parfile):
         """
         param = {}
         repeat_par = {}
-        pfile = open(parfile, 'r')
-        for l in [pl.strip() for pl in pfile.readlines()]:
+        if isinstance(parfile, str):
```

Comment:
The usual python custom is to accept a filename or a file-like object; this is easily added here. In fact, iterating over a file-like object will yield its lines, so this falls directly into the case below.